integration_test/third_party_apps_data/simulacra.go

```diff
@@ -0,0 +1,140 @@
+	}
+	return "", fmt.Errorf("distroFolder() could not find matching folder holding scripts for platform %s", platform)
+}
+
+func setupOpsAgent(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, configFilePath string) error {
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	config, err := confgenerator.UnmarshalYamlToUnifiedConfig(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	if err := agents.SetupOpsAgent(ctx, logger.ToMainLog(), vm, config.String()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getAllReceivers(config *confgenerator.UnifiedConfig) (receivers []string) {
+	for logs := range config.Logging.Receivers {
+		app := config.Logging.Receivers[logs].Type()
+		receivers = append(receivers, app)
+	}
+
+	for logs := range config.Metrics.Receivers {
+		app := config.Metrics.Receivers[logs].Type()
+		receivers = append(receivers, app)
+	}
+	return receivers
+}
+
+// installApps reads an Ops Agent config file and then identifies all the third party apps that need to be installed.
+// The function identifies third party apps to install by checking if any of the receiver types have a
+// corresponding install script in the third_party_apps_data directory.
+// If there is a corresponding install script, then that install script is run on the vm.
+func installApps(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, configFilePath string) error {
+	config, err := confgenerator.MergeConfFiles(ctx, configFilePath, apps.BuiltInConfStructs)
+	if err != nil {
+		return err
+	}
+
+	folder, err := distroFolder(vm.Platform)
+
+	if err != nil {
+		return err
+	}
+
+	receivers := getAllReceivers(config)
+
+	for _, app := range receivers {
+		if scriptContent, err := scriptsDir.ReadFile(path.Join("applications", app, folder, "install")); err == nil {
+			logger.ToMainLog().Printf("Installing %s to VM", app)
+			if _, err := gce.RunScriptRemotely(ctx, logger, vm, string(scriptContent), nil, make(map[string]string)); err != nil {
+				return err
+			} else {
+				logger.ToMainLog().Printf("Done Installing %s", app)
+			}
+		}
+	}
+	return nil
+}
+
+func main() {
+	logger, err := logging.NewDirectoryLogger(path.Join("logs"))
+	if err != nil {
+		log.Default().Fatalf("Failed to setup logger %v", err)
+	}
+	ctx := context.Background()
+	platform := flag.String("platform", "debian-10", "Optional. The OS for the VM. If missing, debian-10 is used.")
+	configFile := flag.String("config_file", "", "Optional. Path to the Ops Agent Config File.")
+	project := flag.String("project", "", "Optional. If missing, the environment variable PROJECT will be used.")
+	zone := flag.String("zone", "", "Optional. If missing, the environment variable ZONE will be used.")
+	name := flag.String("name", fmt.Sprintf("simulacra-vm-instance-%s", uuid.NewString()), "Optional. A name for the instance to be created. If missing, a random name with prefix 'simulacra-vm-instance' will be assigned. ")
```

Comment:
Is `uuid.NewString()` always a static size? Just want to make sure we're not gonna bump up on those VM name size limits.

Reply:
I believe it is a static size based on this doc https://pkg.go.dev/github.com/google/UUID#UUID.String:~:text=String%20returns%20the%20string%20form%20of%20uuid%2C%20xxxxxxxx%2Dxxxx%2Dxxxx%2Dxxxx%2Dxxxxxxxxxxxx%20%2C%20or%20%22%22%20if%20uuid%20is%20invalid.

We could also just use a random int tbh

Reply:
Kept it to the UUID, since it is a static length after all. It seemed to me to be a better design choice than using a random number since it guarantees uniqueness.